client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
To me this is revealing a bigger problem that might need to be addressed: that as soon as you like the comment it disappears (wholly unexpected!)

Is an _undo_ notice the most appropriate response to this problem? What other options could we explore?

For example, what if we kept a comment in the unapproved list after liking it, somehow indicating that now it's approved, but not switching lists? We do this in the notifications client with unread notifications, for example. The background color can indicate that it's not approved, plus the change of buttons. Once we navigate to another list we can stop faking it.

@drw158 are you familiar with this undo behavior?